pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -109,39 +109,39 @@ func (pl *VolumeZone) Filter(ctx context.Context, _ *framework.CycleState, pod *
 		}
 		pvcName := volume.PersistentVolumeClaim.ClaimName
 		if pvcName == "" {
-			return framework.NewStatus(framework.Error, "PersistentVolumeClaim had no name")
+			return framework.NewStatus(framework.UnschedulableAndUnresolvable, "PersistentVolumeClaim had no name")
 		}
 		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
 		if err != nil {
-			return framework.AsStatus(err)
+			return framework.NewStatus(framework.UnschedulableAndUnresolvable, err.Error())
```

Comment:
Can we change this to only return `UnschedulableAndUnresolvable` if the error is of type NotFound?

same for storage class below